### PR TITLE
rewrote singleton objects tour

### PR DESCRIPTION
**Pull request by som-snytt**
Rebased Travis's work.

### CODE REVIEW

**Comment by SethTisue** [comment]
the addition of the `.ruby-version` file is accidental?

File: _tour/singleton-objects.md [hunk 1]
```diff
@@ -10,67 +10,101 @@ num: 13
...
 ```
+The `object Email` contains a factory `fromString` which creates an `Email` instance from a String. We return it as an `Option[Email]` in case of parsing errors.
 
-This illustrates another feature: in the context of `private`, a class and its companion are friends. `object X` can access private members of `class X`, and vice versa. To make a member *really* private to one or the other, use `private[this]`.
+Note: If a class or object has a companion, both must be defined in the same file. To define companions in the REPL, either define them on the same line or enter `:paste` mode.
+
+## Notes for Java programmers ##
 
-For Java convenience, methods, including `var`s and `val`s, defined directly in a singleton object also have a static method defined in the companion class, called a *static forwarder*. Other members are accessible via the `X$.MODULE$` static field for `object X`.
+`static` members in Java are modeled as ordinary members of a companion object in Scala.
```
Comment (jvican) [inline _tour/singleton-objects.md]: @som-snytt I think this is a very important concept, and there are several deletions that explain how people can encode statics in Scala 🤔 Do you think it's reasonable that we emphasize it more in the new section?

**Comment by SethTisue** [comment]
ping @som-snytt, shall we try to get this across the finish line?

**Comment by som-snytt** [comment]
@SethTisue We shall! or maybe go for a field goal.

**Comment by SethTisue** [comment]
no way. touchdown or go home.

File: _tour/singleton-objects.md (hunk 1 above)
Comment (som-snytt) [reply _tour/singleton-objects.md]: Sorry for the late reply. Re-reading the diff, I see that static forwarding is defined on the next line 110. I'm not sure if it's less visible in the diff, or if it really needs more words.

**Comment by som-snytt** [comment]
I did that thing where the snap is to some guy who pitches it forward to the QB.

I have no idea what I'm doing, but I deleted the spurious file from the old contribution and amended my commit text.

I think static forwarders are adequately addressed, but no doubt someone might like to add more words later.

**Comment by SethTisue** [comment]
> I think static forwarders are adequately addressed, but no doubt someone might like to add more words later.

@jvican followup PR if you like

**Comment by SethTisue** [comment]
Homer Zuckerman: There he is!
Arable: That's Som Snytt.
Avery Arable: He's terrific.
Lurvy: He's radiant.
Mrs. Zuckerman: Well, he's clean anyway. That buttermilk certainly helped.

**Comment by som-snytt** [comment]
TIL snytt is norwegian for suckling pig.

**Pull request merged**